### PR TITLE
HOSTEDCP-1971: hypershift: add HO upgrade test

### PR DESCRIPTION
**Pull request by devguyio**
### Description
Add a pre-submit test which runs the hypershift operator upgrade test. 

ref: openshift/hypershift#4999

### CODE REVIEW

**Comment by enxebre** [comment]
/hold
on https://github.com/openshift/hypershift/pull/4999

**Comment by devguyio** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-aws pull-ci-openshift-hypershift-main-e2e-upgrade-tests-aws

File: ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml [hunk 1]
```diff
@@ -127,6 +131,16 @@ tests:
       ENABLE_HYPERSHIFT_CERT_ROTATION_SCALE: "true"
       REQUEST_SERVING_COMPONENT_TEST: "true"
     workflow: hypershift-aws-e2e-external
+- as: e2e-upgrade-tests-aws
```
Comment (enxebre) [inline ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml:134]: can we name this e2e-aws-upgrade-ho to be more consistent with the other names@openshift-ci
e2e-aks, e2e-aws, e2e-aws-4-17, e2e-kubevirt-aws-ovn-reduced...

**Comment by devguyio** [comment]
found a bug, this will use the `hypershift` cli under test for installing HO. Will fix.

File: ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml (hunk 1 above)
Comment (devguyio) [reply ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml]: renamed it to `e2e-aws-upgrade-hypershift-operator`

**Comment by devguyio** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-aws pull-ci-openshift-hypershift-main-e2e-upgrade-tests-aws

**Comment by devguyio** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-aws-upgrade-hypershift-operator

**Comment by devguyio** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-aws-upgrade-hypershift-operator

**Comment by devguyio** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-aws-upgrade-hypershift-operator

**Comment by devguyio** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-aws-upgrade-hypershift-operator

**Comment by devguyio** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-aws-upgrade-hypershift-operator

**Comment by devguyio** [comment]
/pj-rehearse pull-ci-openshift-hypershift-main-e2e-aws-upgrade-hypershift-operator

**Comment by devguyio** [comment]
ci/rehearse/openshift/hypershift/main/e2e-aws

**Comment by devguyio** [comment]
//hold cancel

**Pull request merged**